cmd/network-slirp-binding/domain/domain_test.go

```diff
@@ -0,0 +1,162 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package domain_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	vmschema "kubevirt.io/api/core/v1"
+
+	domainschema "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+
+	"kubevirt.io/kubevirt/cmd/network-slirp-binding/domain"
+)
+
+var _ = Describe("QEMU slirp networking", func() {
+	Context("configure domain spec slirp interface QMEU command line", func() {
+		var testSearchDomain = []string{"dns.com"}
+
+		It("should fail given no pod network", func() {
+			network := vmschema.Network{Name: "secondary", NetworkSource: vmschema.NetworkSource{Multus: &vmschema.MultusNetwork{}}}
+
+			_, err := domain.NewSlirpNetworkConfigurator(nil, []vmschema.Network{network}, nil)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should fail given no slirp iface", func() {
+			network := vmschema.Network{Name: "secondary", NetworkSource: vmschema.NetworkSource{Multus: &vmschema.MultusNetwork{}}}
+			iface := vmschema.Interface{Name: "secondary", InterfaceBindingMethod: vmschema.InterfaceBindingMethod{Bridge: &vmschema.InterfaceBridge{}}}
+
+			_, err := domain.NewSlirpNetworkConfigurator([]vmschema.Interface{iface}, []vmschema.Network{network}, nil)
+			Expect(err).To(HaveOccurred())
+		})
+
+		DescribeTable("should succeed, given",
+			func(iface vmschema.Interface, network vmschema.Network, expectedQEMUCmdArgs []domainschema.Arg) {
+				testDomainSpec := &domainschema.NewMinimalDomain("test").Spec
+
+				expectedDomainSpec := &domainschema.NewMinimalDomain("test").Spec
+				expectedDomainSpec.QEMUCmd = &domainschema.Commandline{QEMUArg: expectedQEMUCmdArgs}
+
+				testMutator, err := domain.NewSlirpNetworkConfigurator([]vmschema.Interface{iface}, []vmschema.Network{network}, testSearchDomain)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testMutator.Mutate(testDomainSpec)).To(Equal(expectedDomainSpec))
+			},
+			Entry("default slirp interface",
+				*vmschema.DefaultSlirpNetworkInterface(),
+				*vmschema.DefaultPodNetwork(),
+				[]domainschema.Arg{
+					{Value: `-netdev`},
+					{Value: `user,id=default,net=10.0.2.0/24,dnssearch=dns.com`},
+					{Value: `-device`},
+					{Value: `{"driver":"e1000","netdev":"default","id":"default"}`},
+				},
+			),
```

Comment:
There is a need to check that given a domain with an existing slirp configuration defined, it does not create another one but keeps the original (or overrides it).

Reply:
With the current implementation additional command line args will be added.
I added a test to reflect this behavior.

In practice, when I tested the new sidecar the generated domain XML had the correct command line args although the `onDefineDomain` hook called twice (no duplicating args).

We can make the slirp plugin smarter up by making it looks into the QMEU command line arg values and act accordingly (not adding args if already exist, add missing args, etc..).but it will require more work and testing.

How about we do that on a follow up PR if we see there is a real issue?

Reply:
> In practice, when I tested the new sidecar the generated domain XML had the correct command line args although the onDefineDomain hook called twice (no duplicating args).

How is this possible? If it is called twice, it is suppose to add twice the configuration.

> How about we do that on a follow up PR if we see there is a real issue?

It is a real issue because you cannot assume it is not called 100 times.

And I still expect to see this expressed in a test.

Reply:
OK, we can consider it as a bug. But please follow up on this.